### PR TITLE
[No JIRA] Add UI component for Icon rendering

### PR DESCRIPTION
**Pull request by k0nserv**
Implement a slightly simpler API for using the icons to render a
`UIImageView`.

The hyptohesis and experience is that icons are most commonly rendered in a single tint with `UIImageView` so this aims to make that use case easier.

### CODE REVIEW

File: Backpack/Icon/Classes/BPKIconView.m [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+@implementation BPKIconView
+
+- (instancetype)initWithIconName:(BPKIconName)iconName size:(BPKIconSize)size {
+    CGSize displaySize = [BPKIcon concreteSizeForIconSize:size];
+
+    self = [super initWithFrame:CGRectMake(0, 0, displaySize.width, displaySize.height)];
+
+    if (self) {
+        self.size = size;
```
Comment (tevelee) [inline Backpack/Icon/Classes/BPKIconView.m]: Can we be explicit about the content mode of the image view? Scale to fit UIViewContentModeScaleAspectFit so it won’t get cropped?

File: Backpack/Icon/Classes/BPKIconView.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+- (instancetype)initWithIconName:(nullable BPKIconName)iconName size:(BPKIconSize)size;
+
+/**
+ * The name of the icon currently being displayed. Setting updates
+ * the displayed icon.
+ */
+@property(nullable, nonatomic, copy) BPKIconName iconName;
+
+- (instancetype)initWithImage:(nullable UIImage *)image __attribute__((unavailable("use `initWithIconName:size:` instead")));
+- (instancetype)initWithImage:(nullable UIImage *)image
```
Comment (tevelee) [inline Backpack/Icon/Classes/BPKIconView.h]: Can you also disable the setImage method for the image property?

File: Backpack/Icon/README.md [hunk 3]
```diff
@@ -2,22 +2,46 @@
...
+
+```objective-c
 #import <Backpack/Color.h>
+#import <Backpack/Icon.h>
 
 UIImage *renderedIcon = [BPKIcon iconNamed:@"flight" color:[BPKolor gray900] size:BPKIconSizeSmall];
 ```
 
 ### Swift
+p
```
Comment (tevelee) [inline Backpack/Icon/README.md]: Typo here

File: Backpack/Icon/Classes/BPKIconView.m (hunk 1 above)
Comment (k0nserv) [reply Backpack/Icon/Classes/BPKIconView.m]: Couldn't hurt

File: Backpack/Icon/Classes/BPKIconView.h (hunk 2 above)
Comment (k0nserv) [reply Backpack/Icon/Classes/BPKIconView.h]: Sounds good

**Pull request merged**